File: level1JS/getPrimeNumbers.js
const getCombinations = function (arr, selectNumber) {
    const results = [];
    if (selectNumber === 1) return arr.map((el) => [el]);


    arr.forEach((fixed, index, origin) => {
        const rest = origin.slice(index + 1);

        const combinations = getCombinations(rest, selectNumber - 1);

        const attached = combinations.map((el) => [fixed, ...el]);

        results.push(...attached);

    });

    return results;
}

const isPrimeNumber = function(num) {
    for(let i = 2; num > i; i++) {
        if(num % i === 0) return false;
    }

    return true;
}

function solution(nums) {
    var answer = 0;

    const addedNumList = [];

    const allNumList = getCombinations(nums, 3);
    console.log(allNumList)
    allNumList.forEach(list => {
        addedNumList.push(list.reduce((acc, v, i) => acc + v));
    })

    addedNumList.forEach(value => {
        if(isPrimeNumber(value)) answer++;
    } )

    return answer;
}

console.log(solution([1,2,3,4]))

File: level1JAVA/src/GetPrimeNumbers.java
class GetPrimeNumbers {
    public int solution(int[] nums) {
        int answer = 0;

        for(int i = 0; i < nums.length; i++) {
            for(int j = i + 1; j < nums.length; j++) {
                for(int k = j + 1; k < nums.length; k++) {
                    int temp = nums[i] + nums[j] + nums[k];
                    if(isPrimeNumber(temp)) answer++;
                }
            }
        }
        return answer;
    }

    public boolean isPrimeNumber(int targetNumber) {
        for(int i = 2; i < targetNumber; i++) {
            if(targetNumber % i == 0) return false;
        }

        return true;
    }

    public static void main(String[] args) {
        GetPrimeNumbers getPrimeNumbers = new GetPrimeNumbers();
        int result = getPrimeNumbers.solution(new int[]{1, 2, 3, 4});
        System.out.println("result = " + result);
    }
}